### ISSUE TITLE
Cannot install without ActiveRecord

### ISSUE THREAD

**Issue by vatsaljain** [simukappu/activity_notification#20]
Hi
I installed the activity_notification gem by editing Gemfile and running bundle install, but whenever i run **bin/rails generate activity_notification:install** in terminal, I get error "_Gem Load Error is: uninitialized constant ActiveRecord_". Also, i get the same error while trying to run rails server or console. When i require active record in application, it gives me error "_Could not load database configuration. No such file - ["config/database.yml"] (RuntimeError)_". Is it possible to install gem without active record as i dont use it anywhere in my app.
I'm using MongoDB without Active Record.
Thanks

**Comment by simukappu** [comment]
Hi,

Currently, `activity_notification` has a dependency on `activerecord >= 4.2.0`.
See https://rubygems.org/gems/activity_notification

However, `activity_notification` allows mongoid models as `Target` and `Notifiable` since version 1.4.0. So, now we can use this gem without any ActiveRecord  models.
We are considering to remove this dependency for the next version.

I would appreciated it if you could try to remove this dependency and run your application.
Thanks!

**Comment by simukappu** [comment]
Hi vatsaljain,

I have just removed *a dependency on activerecord* in **remove_activerecord** branch.
Also removed *ActiveRecord Loading* in the test case which only mongodb is used.

Please try `activity_notification` gem from **remove_activerecord** branch with your application.
```ruby
gem 'activity_notification', git: 'git://github.com/simukappu/activity_notification.git', branch: 'remove_activerecord'
```
Thanks!

**Comment by vatsaljain** [comment]
Thanks !!!
I'll use it without active record.

**Comment by vatsaljain** [comment]
Hi

There's still an issue with the gem. It needs database.yml in the config folder of the app but since i'm using MongoDB, I have mongoid.yml rather than database.yml.

When I tried to install the gem, it gives me error " _Could not load database configuration. No such file - ["config/database.yml"]_ ".

Thanks

**Comment by vatsaljain** [comment]
Hi

Its still giving the same error as before when I try to install it.
Error:

"_ Gem Load Error is: uninitialized constant ActiveRecord
Backtrace for gem load error is:
/home/vatsal/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/activity_notification-1.4.0/lib/activity_notification/models.rb:20:in `<top (required)>'_"

Seems like there's still a dependency in models.rb

Thanks

**Comment by simukappu** [comment]
Hi vatsaljain,

Thank you for your testing. **models.rb line 20** was also updated in removing a *dependency on activerecord*.

Before
```ruby
ActiveRecord::Base.class_eval { include ActivityNotification::Models }
```
After
```ruby
ActiveRecord::Base.class_eval { include ActivityNotification::Models } if defined?(ActiveRecord::Base)
```
Actually **Gem Load Error is: uninitialized constant ActiveRecord** was thrown in *Before* case. Now we can run the test app without "config/database.yml" in *After* case.
Could you share sample app as a small case of your application? I will investigate it.

Thanks

**Comment by vatsaljain** [comment]
Thanks, it installing now without active record and database.yml file.

**Comment by simukappu** [comment]
I have just released **activity_notification 1.4.1** including this removal of dependency on *activerecord*.
Thank you for your support!
